### PR TITLE
[chore] enforce validation of codeowners membership

### PR DESCRIPTION
**Pull request by atoulme**
**Description:**
This adds a validation to our codeowners by forcing us to review and manage the codeowners registered with the OpenTelemetry github organization.

### CODE REVIEW

**Comment by atoulme** [comment]
This PR is blocked by #24637

File: cmd/githubgen/allowlist.txt
Comment (mx-psi) [inline cmd/githubgen/allowlist.txt:22]: Instead of doing this, why not ignore any usernames with the `open-telemetry/` prefix on it? And fail for any usernames with a `/` on it that don't start with `open-telemetry` to exclude teams from other orgs

File: cmd/githubgen/main.go
Comment (mx-psi) [inline cmd/githubgen/main.go:87]: Why not read these at runtime? Recompiling the app every time we change these files doesn't seem ideal and makes us have to leave the app on this repository instead of eventually moving it to opentelemetry-go-build-tools

File: cmd/githubgen/main.go
Comment (mx-psi) [inline cmd/githubgen/main.go:285]: This is likely easier to maintain if we use something like https://github.com/google/go-github. Not necessary to do this on this PR, but we should open an issue about it if we don't use it

File: cmd/githubgen/members.txt
Comment (mx-psi) [inline cmd/githubgen/members.txt:88]: Same as in previous comment, I think we should exclude teams from this tool, since we can easily identify them through the `/`

File: cmd/githubgen/main.go
Comment (atoulme) [reply cmd/githubgen/main.go:87]: It's not in my scope to move this code to opentelemetry-go-build-tools. This code was definitely not designed to be portable, and maybe someone else can work on this. The code runs with a `go run` command right now in the Makefile, effectively embedding the files at the time.

File: cmd/githubgen/allowlist.txt
Comment (atoulme) [reply cmd/githubgen/allowlist.txt:22]: That's a good move, we can allow any open telemetry team for sure.

File: cmd/githubgen/main.go
Comment (atoulme) [reply cmd/githubgen/main.go:288]: I have filed https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/24813 thanks!

File: cmd/githubgen/main.go
Comment (mx-psi) [reply cmd/githubgen/main.go:87]: I am not saying to move this to opentelemetry-go-build-tools now, I am just saying that we should not make it harder to eventually move it by embedding the files. Even if we don't ever move it, embedding the files is probably confusing for people editing said files

Comment (atoulme) [reply cmd/githubgen/main.go]: OK it's now readable and we have an argument to make this configurable

File: cmd/githubgen/main.go
Comment (atoulme) [reply cmd/githubgen/main.go]: I've changed the code to use this lib, and we should be good now. Thanks for the suggestion.

File: cmd/githubgen/allowlist.txt
Comment (djaglowski) [inline cmd/githubgen/allowlist.txt:20]: This doesn't have to be in this PR but would it be difficult to cross reference the allow list with the member list? I see that mrsillydog is listed in both and is a member. It would be nice if we had a mechanism to ensure we are keeping this list to a minimum.

**Review by djaglowski — APPROVED** [review]

File: go.mod
Comment (codeboten) [inline go.mod:171]: why this change?

Comment (atoulme) [reply go.mod:171]: Most likely because the module has its own go.mod, so this dependency is no longer needed to be applied from the root go.mod.

File: pkg/winperfcounters/metadata.yaml
Comment (atoulme) [reply pkg/winperfcounters/metadata.yaml:3]: I just might.

Comment (atoulme) [reply pkg/winperfcounters/metadata.yaml:3]: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/25848

File: cmd/githubgen/allowlist.txt
Comment (atoulme) [reply cmd/githubgen/allowlist.txt]: I ended up pushing that change into this PR.

File: pkg/winperfcounters/metadata.yaml
Comment (codeboten) [reply pkg/winperfcounters/metadata.yaml:3]: @atoulme thank you!

File: cmd/githubgen/main.go
Comment (MovieStoreGuy) [inline cmd/githubgen/main.go:146]: Any reason for the magic number here?

File: cmd/githubgen/main.go
Comment (MovieStoreGuy) [inline cmd/githubgen/main.go:256]: Could it be possible to use maps instead?

File: cmd/githubgen/main.go
Comment (MovieStoreGuy) [inline cmd/githubgen/main.go:259]: Is it worth passing these values in?

**Review by MovieStoreGuy — COMMENTED** [review]
Where are the tests?

File: cmd/githubgen/main.go
Comment (atoulme) [reply cmd/githubgen/main.go:259]: yes, otherwise it fails because the github token is not set.

File: cmd/githubgen/main.go
Comment (atoulme) [reply cmd/githubgen/main.go:256]: If we must? Do we care about performance all that much for this code?

File: cmd/githubgen/main.go
Comment (atoulme) [reply cmd/githubgen/main.go:146]: I picked something. Should I just not try to set a value?

File: cmd/githubgen/allowlist.txt
Comment (atoulme) [reply cmd/githubgen/allowlist.txt:1]: allowlist of github usernames who are not currently code owners and yet allowed to be kept in while we resolve https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/20868

File: cmd/githubgen/main.go
Comment (atoulme) [reply cmd/githubgen/main.go]: done

File: cmd/githubgen/main.go
Comment (atoulme) [reply cmd/githubgen/main.go]: OK removed :)

**Comment by atoulme** [comment]
> Where are the tests?
This runs against a Github API and runs against all components at CI time. 100% of the code is used as part of each run.
"Works on github runner"

**Review by dmitryax — APPROVED** [review]

**Pull request merged**
